CRM/Import/ImportProcessor.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * Class CRM_Import_ImportProcessor.
+ *
+ * Import processor class. This is intended to provide a sanitising wrapper around
+ * the form-oriented import classes. In particular it is intended to provide a clear translation
+ * between the saved mapping field format and the quick form & parser formats.
+ *
+ * In the first instance this is only being used in unit tests but the intent is to migrate
+ * to it on a trajectory similar to the ExportProcessor so it is not in the tests.
+ */
+class CRM_Import_ImportProcessor {
+
+  /**
+   * An array of fields in the format used in the table civicrm_mapping_field.
+   *
+   * @var array
+   */
+  protected $mappingFields = [];
+
+  /**
+   * Get contact type being imported.
+   *
+   * @var string
+   */
+  protected $contactType;
+
+  /**
+   * @return string
+   */
+  public function getContactType(): string {
+    return $this->contactType;
+  }
+
+  /**
+   * @param string $contactType
+   */
+  public function setContactType(string $contactType) {
+    $this->contactType = $contactType;
+  }
+
+  /**
+   * @return array
+   */
+  public function getMappingFields(): array {
+    return $this->mappingFields;
+  }
+
+  /**
+   * @param array $mappingFields
+   */
+  public function setMappingFields(array $mappingFields) {
+    $this->mappingFields = CRM_Utils_Array::rekey($mappingFields, 'column_number');
+    ksort($this->mappingFields);
+  }
+
+  /**
+   *
+   */
+  public function getFieldNames() {
+    return CRM_Utils_Array::collect('name', $this->getMappingFields());
+  }
+
+  /**
+   * Get an instance of the importer object.
+   *
+   * @return CRM_Contact_Import_Parser_Contact
+   */
+  public function getImporterObject() {
+    $importer = new CRM_Contact_Import_Parser_Contact($this->getFieldNames());
```

Comment:
Actually we want to pass more params here but I wanted to do that as a follow up